### ISSUE TITLE
Async non-blocking call to datastore?

### ISSUE THREAD

**Issue by susanlinsfu** [googleapis/google-cloud-java#493]
Will there be any support for async non-blocking calls to the datastore?

**Comment by mziccard** [comment]
Hi @susanlinsfu thanks for your interest.
I think that support for async non-blocking calls will be added once we migrate to gRPC for the transport layer. But I don't have a date to tell you yet. Maybe @jgeewax and @aozarov have more information than I do.

**Comment by aozarov** [comment]
Yes, as @mziccard said, we do plan to provide async calls to the Datastore API but
that would be part of the transition to gRPC which will happen early next year (no exact date yet).

**Comment by susanlinsfu** [comment]
gRPC has now come to the datastore. Is there a date yet when async features will be released?

**Comment by aozarov** [comment]
It is in the roadmap but unfortunately no dates yet.

**Comment by susanlinsfu** [comment]
Thanks for the very quick response! If it means anything at all I am really pushing for this feature as spinning up threads for every request really is killing performance! :)

**Comment by aozarov** [comment]
Until then you should consider grouping operation when possible using batches.

**Comment by garrettjonesgoogle** [comment]
Closing out this issue - even if we left it open, we might overlook updating it when we add gRPC support to datastore. It's not scheduled yet even now.

**Comment by lesv** [comment]
This is a very important request.  Not sure how you want to track asks like this.

**Comment by garrettjonesgoogle** [comment]
I think I will re-open this one due to the attention on it.

**Comment by lesv** [comment]
Perhaps we can incorporate parts of [Spotify's library](https://github.com/spotify/async-datastore-client).

**Comment by garrettjonesgoogle** [comment]
This is blocked on support for gRPC in app engine.

**Comment by lesv** [comment]
We've made it work, I'm going to test again, but I can do gRPC in GAE Std 7.

**Comment by bmenasha** [comment]
GRPC functions on Java8 Standard  can we please move forward with using GRPC as transport for Datastore?   Thanks

**Comment by garrettjonesgoogle** [comment]
I have removed the blocking label since grpc can be used on GAE Java 8 standard.

**Comment by danoscarmike** [comment]
This has been added to our feature backlog: https://github.com/GoogleCloudPlatform/google-cloud-java/wiki/Feature-backlog. This issue will be closed but is linked in the backlog and can continue to be used for comment and discussion.

**Comment by chbatey** [comment]
Any update on when this will be implemented?

**Comment by iozozturk** [comment]
Hey team, any news?

**Comment by YashchenkoN** [comment]
hey, any plans to make async API for Datastore?

**Comment by davidhardwickbc** [comment]
Any update here?  Seems odd to not have gRPC for database calls, the performance difference between HTTP and gRPC is obvious.
